x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/operator/AbstractPageMappingToIteratorOperator.java

```diff
@@ -154,24 +172,39 @@ public static class Status implements Operator.Status {
         private final long processNanos;
         private final int pagesReceived;
         private final int pagesEmitted;
+        private final int rowsReceived;
+        private final int rowsEmitted;
 
-        public Status(long processNanos, int pagesProcessed, int pagesEmitted) {
+        public Status(long processNanos, int pagesProcessed, int pagesEmitted, int rowsReceived, int rowsEmitted) {
             this.processNanos = processNanos;
             this.pagesReceived = pagesProcessed;
             this.pagesEmitted = pagesEmitted;
+            this.rowsReceived = rowsReceived;
+            this.rowsEmitted = rowsEmitted;
         }
 
         protected Status(StreamInput in) throws IOException {
             processNanos = in.readVLong();
             pagesReceived = in.readVInt();
             pagesEmitted = in.readVInt();
+            if (in.getTransportVersion().onOrAfter(TransportVersions.ESQL_PROFILE_ROWS_PROCESSED)) {
+                rowsReceived = in.readVInt();
+                rowsEmitted = in.readVInt();
+            } else {
+                rowsReceived = 0;
+                rowsEmitted = 0;
+            }
         }
 
         @Override
         public void writeTo(StreamOutput out) throws IOException {
             out.writeVLong(processNanos);
             out.writeVInt(pagesReceived);
             out.writeVInt(pagesEmitted);
+            if (out.getTransportVersion().onOrAfter(TransportVersions.ESQL_PROFILE_ROWS_PROCESSED)) {
+                out.writeVInt(rowsReceived);
+                out.writeVInt(rowsEmitted);
```

Comment:
I wonder if we should do a `long` here. 2 billion rows.... Let's presume we're shoving them in a STATS and reading them fast fast, like, 10 nanoseconds a row. That'd take.... uses python as a calculator..... 21.5 seconds.... Oh. Well. I think these should be `long`s.

Even if it takes 100 nanoseconds a row. Or a microsecond. We're only talking about 35 minutes of work to flip the sign.

Reply:
Huh, absolutely. Changing it!

Reply:
I'm wondering if my math is right. I had sort of expected max int to be big enough for longer than 21.5 seconds of turbo rows.

Reply:
But still, rows should probably `long`.

Another way to think of this is that lucene indices max out at max int documents. And we can process more than one lucene index in these operators.

Reply:
Changed to long, and tested in the benchmarks with +4B rows